pkg/platform/api/api.go

```diff
@@ -0,0 +1,211 @@
+package api
+
+import (
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"time"
+
+	"github.com/bottlerocket-os/bottlerocket-update-operator/pkg/logging"
+
+	"github.com/pkg/errors"
+)
+
+const bottlerocketAPISock = "/run/api.sock"
+const minimumRequiredOSVer = "0.3.4"
+
+type updateState string
+
+const (
+	Idle      updateState = "Idle"
+	Available updateState = "Available"
+	Staged    updateState = "Staged"
+	Ready     updateState = "Ready"
+)
+
+type UpdateImage struct {
+	Arch    string `json:"arch"`
+	Version string `json:"version"`
+	Variant string `json:"variant"`
+}
+
+func (ui *UpdateImage) Identifier() interface{} {
+	return ui.Version
+}
+
+type StagedImage struct {
+	Image      UpdateImage `json:"image"`
+	NextToBoot bool        `json:"next_to_boot"`
+}
+
+type updateCommand string
+
+const (
+	refresh    updateCommand = "refresh"
+	prepare    updateCommand = "prepare"
+	activate   updateCommand = "activate"
+	deactivate updateCommand = "deactivate"
```

Comment:
I was a bit confused seeing these inline, can you add `command` to these consts?

```suggestion
	commandRefresh    updateCommand = "refresh"
	commandPrepare                  = "prepare"
	commandActivate                 = "activate"
	commandDeactivate               = "deactivate"
```